pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -260,3 +272,35 @@ func (r *Reconciler) installPreflight(ctx context.Context, client client.Client,
 
 	return nil
 }
+
+//nolint:gocyclo
+func (r *Reconciler) reconcileSpecialCases(ctx context.Context, c client.Client, yaml []byte) error {
+	objs, err := clientutil.YamlToClientObjects(yaml)
+	if err != nil {
+		return err
+	}
+
+	currentDS, currentDeployment, currentSvc, err := getObjects(ctx, c, cilium.DaemonSetName, cilium.DeploymentName)
```

Comment:
do we need this helper?
just trying to simplify, could we just get the 3 objects separately?

Reply:
We don't need it anymore since we're gonna use Client.Update instead of MergeFrom